src/Workspaces/Remote/Core/Telemetry/ApiUsageIncrementalAnalyzerProvider.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.CodeAnalysis.Remote.Telemetry
+{
+    /// <summary>
+    /// Creates an <see cref="IIncrementalAnalyzer"/> that collects Api usage information from metadata references
+    /// in current solution.
+    /// </summary>
+    [ExportIncrementalAnalyzerProvider(nameof(ApiUsageIncrementalAnalyzerProvider), new[] { WorkspaceKind.RemoteWorkspace }), Shared]
+    internal sealed class ApiUsageIncrementalAnalyzerProvider : IIncrementalAnalyzerProvider
+    {
+        [ImportingConstructor]
+        public ApiUsageIncrementalAnalyzerProvider()
+        {
+        }
+
+        public IIncrementalAnalyzer CreateIncrementalAnalyzer(Workspace workspace)
+        {
+            return new Analyzer();
+        }
+
+        private sealed class Analyzer : IIncrementalAnalyzer
+        {
+            // maximum number of symbols to report per project.
+            private const int Max = 2000;
+
+            private const string EventName = "vs/compilers/api";
+            private const string PropertyName = "vs.compilers.api.pii";
+
+            private readonly HashSet<ProjectId> _reported = new HashSet<ProjectId>();
```

Comment:
should this be static?  what happens if the incremental anlayzer gets created again?  we'd go and reanalyze everything in the solution right?

Reply:
incremental analyzer doesn't get recreated and even if we report it twice due to unknown reason, that won't be end of day.